pokemongo_bot/cell_workers/sniper.py

```diff
@@ -311,60 +325,70 @@ def snipe(self, pokemon):
-            # Backup position before anything
-            last_position = self.bot.position[0:2]
-
-            # Teleport, so that we can see nearby stuff
-            self.bot.hb_locked = True
-            self._teleport_to(pokemon)
-
-            # If social is enabled and if no verification is needed, trust it. Otherwise, update IDs!
-            verify = not pokemon.get('encounter_id') or not pokemon.get('spawn_point_id')
-            exists = not verify and self.mode == SniperMode.SOCIAL
-            success = exists
-
-            # If information verification have to be done, do so
-            if verify:
-                seconds_since_last_check = time.time() - self.last_cell_check_time
-
-                # Wait a maximum of MIN_SECONDS_ALLOWED_FOR_CELL_CHECK seconds before requesting nearby cells
-                if (seconds_since_last_check < self.MIN_SECONDS_ALLOWED_FOR_CELL_CHECK):
-                    time.sleep(self.MIN_SECONDS_ALLOWED_FOR_CELL_CHECK - seconds_since_last_check)
-
-                nearby_pokemons = []
-                nearby_stuff = self.bot.get_meta_cell()
-                self.last_cell_check_time = time.time()
-
-                # Retrieve nearby pokemons for validation
-                nearby_pokemons.extend(nearby_stuff.get('wild_pokemons', []))
-                nearby_pokemons.extend(nearby_stuff.get('catchable_pokemons', []))
-
-                # Make sure the target really/still exists (nearby_pokemon key names are game-bound!)
-                for nearby_pokemon in nearby_pokemons:
-                    nearby_pokemon_id = nearby_pokemon.get('pokemon_data', {}).get('pokemon_id') or nearby_pokemon.get('pokemon_id')
-
-                    # If we found the target, it exists and will very likely be encountered/caught (success)
-                    if nearby_pokemon_id == pokemon.get('pokemon_id', 0):
-                        exists = True
-                        success = True
-
-                        # Also, if the IDs arent valid, override them (nearby_pokemon key names are game-bound!) with game values
-                        if not pokemon.get('encounter_id') or not pokemon.get('spawn_point_id'):
-                            pokemon['encounter_id'] = nearby_pokemon['encounter_id']
-                            pokemon['spawn_point_id'] = nearby_pokemon['spawn_point_id']
-                        break
-
-            # If target exists, catch it, otherwise ignore
-            if exists:
-                self._log('Yay! There really is a wild {} nearby!'.format(pokemon.get('pokemon_name')))
-                self._teleport_back_and_catch(last_position, pokemon)
+            # Have we already tried this pokemon?
+            if not hasattr(self.bot,'sniper_unique_pokemon'):
+                self.bot.sniper_unique_pokemon = []
+            
+            # Check if already in list of pokemon we've tried
+            uniqueid = self._build_unique_id(pokemon)
+            if self._is_cached(uniqueid):
```

Comment:
is_snipeable currently lumps everything together as a general fail. I've removed one of those for clarity.